alpha/apps/kaltura/lib/storage/kFileSyncUtils.class.php

```diff
@@ -908,7 +908,14 @@ public static function getReadyFileSyncForKey ( FileSyncKey $key , $fetch_from_r
 			}
 		}
 
+		$periodicStorageIds = array();
+		if($fetch_from_remote_if_no_local)
+		{
+			$periodicStorageIds = kStorageExporter::getPeriodicStorageIdsByPartner($key->partner_id);
```

Comment:
At this point you should have only sort the list of file_sync_list where the local DC remains first un touched, and the non local should be order where the periodic is first.
file_sync_list = sortFileSyncList($file_sync_list,$priodicRemoteStorageIds)

All the same logic remains exactly the same